lib/phlex/component.rb

```diff
@@ -6,7 +6,12 @@ module Phlex
   class Component
     module Overrides
       def initialize(*args, **kwargs, &block)
+        if block_given? && !block.binding.receiver.is_a?(Block)
+          block = Block.new(self, &block)
+        end
```

Comment:
We need to ensure these blocks are Phlex::Blocks here. This doesn't mean we can remove the Phlex::Block coercion from the other context methods, since it's important to coerce immediately to capture the right context for the Phlex::Block.